controllers/articles.controllers.js

```diff
@@ -33,22 +34,39 @@ exports.getArticleById = (req, res, next) => {
 };
 exports.getArticleComments = (req, res, next) => {
   const { article_id } = req.params;
-  selectArticleById(article_id)
-    .then((article) => {
-      if (!article) {
-        return res
-          .status(404)
-          .send({ msg: `Unable to find Article by ID - ${article_id}` });
-      }
-      selectArticleComments(article_id)
-        .then((comments) => {
-          res.status(200).send({ comments });
-        })
-        .catch((err) => {
-          next(err);
-        });
+  selectArticleById(article_id).then((article) => {
+    if (!article) {
+      return res
+        .status(404)
+        .send({ msg: `Unable to find Article by ID - ${article_id}` });
+    }
+    selectArticleComments(article_id)
+      .then((comments) => {
+        res.status(200).send({ comments });
+      })
+      .catch((err) => {
+        next(err);
+      });
+  });
+};
+
+exports.postArticleComment = (req, res, next) => {
+  const { article_id } = req.params;
+  const { username, body } = req.body;
+  if (!username || !body) {
+    return res.status(400).send({ msg: "Missing required fields" });
+  }
+  if (typeof username !== "string" || typeof body !== "string") {
+    return res.status(400).send({ msg: "Invalid data inputted" });
```

Comment:
Some of these errors might be rejected by SQL. It's best to generate an error and see what happens first before handling the error